ably/realtime_conn_spec_test.go

```diff
@@ -144,12 +144,17 @@ func TestRealtimeConn_RTN15b(t *testing.T) {
 	}
 
 	var metaList []*meta
-
+	gotDial := make(chan chan struct{})
 	app, client := ablytest.NewRealtimeClient(&ably.ClientOptions{
 		NoConnect: true,
 		Dial: func(protocol string, u *url.URL) (proto.Conn, error) {
 			m := &meta{dial: u}
 			metaList = append(metaList, m)
+			if len(metaList) > 1 {
```

Comment:
What's the purpose of this condition? Also, won't it be always true, given we're just appending to `metaList` in the line just above this one?

Reply:
Dial is called twice, and the first time we don't want to block the channel. We want to block with `goOn` channel when dialling the second time where `len(metaList)==2`